backend/utils/fileUpload.ts

```diff
@@ -0,0 +1,113 @@
+import {Request} from 'express';
+import multer from 'multer';
+import path from 'path';
+
+// Configure the file storage
+const storage = multer.diskStorage({
+    destination: (req: Request, file, cb) => {
+        cb(null, 'uploads'); // 'uploads' specified in index.ts
+    },
+    // concatenate hyphenated date string to ensure unique file name
+    filename: (req: Request, file, cb) => {
+        // Limit the input file name size to 50 characters
+        const maxFilenameSize = 50;
+
+        // Check if the original filename exceeds the limit
+        if (file.originalname.length > maxFilenameSize) {
+            const error = new Error('Filename too long');
+            cb(error, '');
+        } else {
+            // Generate the filename by adding date string to original name
+            const generatedFilename =
+                new Date().toISOString().replace(/:/g, '-') +
+                '-' +
+                file.originalname;
+            cb(null, generatedFilename);
+        }
+    },
+});
+
+// Configure file size limits
+const limits = {
+    fileSize: 1024 * 1024 * 2, // 2 MB limit
+    files: 1, // 1 file upload limit
+};
+
+// Configure the allowed file formats
+// Validates both the file extension and type for security
+const fileFilter = (req: Request, file: any, cb: any) => {
+    try {
+        if (!file || !file.originalname) {
+            throw new Error('Invalid file object');
+        }
+        const allowedExtensions = new RegExp(/\.(jpg|png|jpeg|gif)$/i);
```

Comment:
Do we want to allow users to uploads gifs? I think we shouldn't. 
Either need to remove gif from allowedExtensions regex or add image/gif to allowedMimeTypes

Reply:
this file isn't being used rn. We can look to get the validation working with the implementation in another PR.  I just didn't want to lose the code you wrote. we likely shouldn't let gifs. I'll remove it